### PR TITLE
Remove bottle_version

### PR DESCRIPTION
**Pull request by xu-cheng**
This requires #40132 merged first.

cc @Homebrew/owners @bfontaine @DomT4

### CODE REVIEW

File: Library/Homebrew/bottles.rb [hunk 1]
```diff
@@ -39,11 +38,18 @@ def bottle_tag
...
+  receipt_file = Utils.popen_read("tar", "-xOzf", bottle_file, receipt_file_path)
+  name = receipt_file_path.split("/").first
+  tap = Tab.from_tabfile_content(receipt_file, "#{bottle_file}/#{receipt_file_path}").tap
+
+  if tap.nil? || tap == "Homebrew/homebrew"
+    full_name = name
+  else
+    full_name = "#{tap.sub("homebrew-", "")}/#{name}"
+  end
+  [name, full_name]
```
Comment (mikemcquaid) [inline Library/Homebrew/bottles.rb]: I don't think this should return two arguments without changing the function name to indicate that.

Comment (xu-cheng) [inline Library/Homebrew/bottles.rb]: The name is already changed. Maybe a better name?

Comment (mikemcquaid) [inline Library/Homebrew/bottles.rb]: `bottle_both_formula_names` or something

Comment (xu-cheng) [inline Library/Homebrew/bottles.rb]: I finally choose `bottle_resolve_formula_names`.

Comment (mikemcquaid) [inline Library/Homebrew/bottles.rb]: Still a bit yuck but can't think of anything better so :+1:

**Comment by xu-cheng** [comment]
Any other comments for this PR?

**Comment by mikemcquaid** [comment]
This is great, love the approach.

**Comment by xu-cheng** [comment]
Tweaked a bit on the method name. I think it's good to merge.

Notice after this is merged, old `*.bottle.rb` wouldn't be recognized. So I would like to pick a time when no jobs running on ci-bot to merge this.

**Comment by mikemcquaid** [comment]
Ok, cool. How does the `.bottle.rb` file change? Can you paste a before/after?

**Comment by xu-cheng** [comment]
> How does the .bottle.rb file change? Can you paste a before/after?

I just add an extra comment line to indicate the formula name. Like this:
```
$ cat tree-1.7.0.yosemite.bottle.rb
# tree
  bottle do
    prefix "/Users/xucheng/Coding/Github/homebrew"
    cellar "/Users/xucheng/Coding/Github/homebrew/Cellar"
    revision 1
    sha256 "ff6596f918d28daa58903a49409767c30fa7e3aef9cc1bc561ce0f2f3ce1b840" => :yosemite
  end
```

File: Library/Homebrew/bottles.rb [hunk 2]
```diff
@@ -39,11 +38,19 @@ def bottle_tag
   end
 end
 
-def bottle_filename_formula_name filename
-  path = Pathname.new filename
-  version = BottleVersion.parse(path).to_s
-  basename = path.basename.to_s
-  basename.rpartition("-#{version}").first
+def bottle_resolve_formula_names bottle_file
+  receipt_file_path = Utils.popen_read("tar", "-tzf", bottle_file, "*/*/INSTALL_RECEIPT.json").chomp
```
Comment (xu-cheng) [inline Library/Homebrew/bottles.rb]: For gnu-tar (hence linuxbrew), it requires an extract flag (`--wildcards`) to make it work. Should I add os detection here,   or put it only in linuxbrew?

Comment (mikemcquaid) [inline Library/Homebrew/bottles.rb]: Only in Linuxbrew, I think.

**Comment by mikemcquaid** [comment]
@xu-cheng cool.

**Pull request closed without merging**